### PR TITLE
More fixes for the format strings fallout

### PR DESCRIPTION
**Pull request by lrusak**
Many of these changes were not obvious because they were used indirectly by `StringUtils::Format`.

This should fix many of the found issues so far but there is likely more to come (especially with the SQL statements). I will look into the VideoDatabase problems next.

### CODE REVIEW

File: xbmc/interfaces/builtins/GUIBuiltins.cpp [hunk 1]
```diff
@@ -306,7 +306,8 @@ static int Screenshot(const std::vector<std::string>& params)
     {
       if (XFILE::CDirectory::Exists(strSaveToPath))
       {
-        std::string file = CUtil::GetNextFilename(URIUtils::AddFileToFolder(strSaveToPath, "screenshot%05d.png"), 65535);
+        std::string file = CUtil::GetNextFilename(
+            URIUtils::AddFileToFolder(strSaveToPath, "screenshot{:05}.png"), 65535);
```
Comment (Montellese) [inline xbmc/interfaces/builtins/GUIBuiltins.cpp:310]: Not related to this PR but why do we have the same logic here and in `CScreenShot::TakeScreenshot()`?

**Review by Montellese — APPROVED** [review]
Looks good.

**Review by thexai — APPROVED** [review]

**Comment by Montellese** [comment]
As mentioned before: Be careful with string formatting in classes derived from `CDatabase` because there we must use `PrepareSQL()` which uses the old style formatting.

**Comment by Montellese** [comment]
I've grepped through the code and found the following files which still contain old style formatting:
```
xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecFFmpeg.cpp
xbmc/cores/VideoPlayer/DVDDemuxSPU.cpp
xbmc/games/GameUtils.cpp
xbmc/guilib/GUISliderControl.cpp
xbmc/interfaces/json-rpc/SettingsOperations.cpp
xbmc/interfaces/legacy/InfoTagPicture.cpp
xbmc/network/AirPlayServer.cpp
xbmc/peripherals/dialogs/GUIDialogPeripheralSettings.cpp
xbmc/settings/dialogs/GUIDialogSettingsManualBase.cpp
xbmc/settings/dialogs/GUIDialogSettingsManualBase.h
xbmc/utils/StringUtils.cpp
xbmc/video/dialogs/GUIDialogAudioSettings.cpp
```
and I haven't checked the database files yet.

**Comment by Montellese** [comment]
@lrusak: 6a433b1bd258b1edb3be97094274ca732a52a134 takes care of all the ones that I found by grepping for `%[-0-9.]*[a-zA-Z]{1,2}`. Unfortunately the resulting list is very long so I went through it by hand... This also includes the ones related to settings and database queries as already reported by others.

**Comment by lrusak** [comment]
@Montellese thanks very much for taking the time to look into this! I've picked your commit and pushed it here.

**Review by garbear — APPROVED** [review]

**Comment by DaveTBlake** [comment]
Linking this fix back to the PR that caused the regression https://github.com/xbmc/xbmc/pull/19796

> This should fix many of the found issues so far but there is likely more to come (especially with the SQL statements). I will look into the VideoDatabase problems next.

This PR does fix the issues in music smart playlists (with certain rules), and I don't think there are any other impacting the music database. 

Worth noting that old style formatting arguments are correct in some places as they are still used by `PrepareSQL()` when building SQL statements. The  prepare methods in the respective MySQL and SQLite db wrappers use external routines that expect this style (with certain modifications).

**Comment by howie-f** [comment]
noticed some more fallout in the "Video settings" dialog.
Audio, Subtitles, etc. are not affected as far as i can see.
hope it is ok to place it here for the record only.
screenshot00000

**Pull request merged**